### DISCUSSION TITLE
Raretés variables

### DISCUSSION THREAD

**Discussion by Easygamer01s** [Crownicles/Crownicles#1615] [Suggestions fermées]
Hello :)

En fonction des prédictions du mini event lié à l'api de la NASA, certaines catégories de raretés de drop sont supprimées.
Par exemple les objets "peu commun" disparaissent du shop les jours de pleine lune/éclipse lunaire..
Les probabilités associés aux autres classes sont donc augmentés, conditionnées par la rareté des objets supprimés..
Dans l'exemple des "peu commun", les raretés sont multipliés par 4/3. Les légendaires dropent avec 0,3059 % au lieu de 0,23%
Il pourrait y avoir des mauvais jours où les raretés type épique, spécial etc sont supprimées..
Je pense que si l'on fait en mesure que le shop ne devient pas rentable, ça pourrait permettre de lier les astres à l'évolution des joueurs :)

**Comment by voltou91** [comment]
Salut,
 je t'avoue que personnellement je n'aime pas trop. Je veux dire empêcher de looter des items à cause des astres je trouve que ça fait trop. Je me doute que tu veux augmenter les proba pour looter de bon item mais je pense pas que ce soit une bonne solution.
De plus au niveau technique je ne pense pas que ce soit ouf pour les performances du bot, car les taux de loot sont des constantes. Et faire ce que tu dit nous obligerait à recalculer à chaque fois les probas en fonction des astres.
Bonne journée :)

**Reply by Feiryn** [reply]
Ce n'est pas ça qui va impacter les perfs du bot

**Reply by Easygamer01s** [reply]
Merci pour ton retour :) 
L'objectif n'est vraiment pas de faire sauter le système de raretés actuel, un légendaire resterait un objet très rare.. 
Je trouve simplement dommage que l'api et tout le code ait été dev juste pour un mini event :) et je trouvais cool l'idée d'accumuler de la tune jusqu'a la prochaine éclipse :)

**Comment by BastLast** [comment]
Hello, je ne pense pas que cela soit une bonne idée pour plusieurs raisons :

- Faire l'équilibrage de ce système me parait assez compliqué.
- Programmer cela ne necessiterait pas simplement de copier coller le code du mini event ou de le réutiliser vu que ce dernier n'a pas été conçu avec cette évolution en tête, ça resterait une opération assez longue
- La logique de pleine lune deviendrait centrale à draftbot et ce n'est pas une mécanique que j'ai envie de mettre autant en avant

Cependant, nous pensons réutiliser les cycle de lune dans un futur mini-évènement à long terme, donc il est possible que nous donnions plus d'importance à la mécanique de pleine lune mais pas rapidement et pas sur un truc aussi central que le loot d'objet

**Reply by BastLast** [reply]
je me permet de déplacer cette idée dans la catégorie rejettée suite à cette réponse
